### PR TITLE
Tests for input added

### PR DESCRIPTION
**Pull request by Bhumika-Kothwal**
1. validate_input function added in preprocessy/input/_read.py
         -> so that non-string type values passed to file_name error can be caught 
2. test for the same is added in tests/test_input.py

### CODE REVIEW

File: preprocessy/input/_read.py [hunk 1]
```diff
@@ -2,13 +2,20 @@
...
     def __init__(self, file_name):
         self.excel_extensions = ["xls", "xlsx", "xlsm", "xlsb", "odf", "ods", "odt"]
-        self.file_name = file_name
+        self._validate_input(file_name)
         self.__read_file()
         self.summary, self.stats = self.__read_summary()
 
+    def _validate_input(self, file_name):
+        if type(file_name) is not str:
+            raise TypeError("Argument file_name should be of str type")
```
Comment (Saif807380) [inline preprocessy/input/_read.py:15]: Can you also show what was the type of `file_name` received? So the error would look like

> Argument file_name should be of type str. Received xyz

You can use `f strings` for this.

File: tests/test_input.py [hunk 2]
```diff
@@ -5,6 +5,13 @@
...
 
 class TestReader:
+    def test_incorrect_file_name_type(self):
+        with pytest.raises(TypeError):
+            reader = ReadData(None)
+
+        with pytest.raises(TypeError):
+            reader = ReadData(file_name=["datasets/encoding/test.csv"])
+
     def test_file_name(self):
```
Comment (Saif807380) [inline tests/test_input.py:15]: `test_incorrect_file_type` looks like a better name for this test.

**Review by Saif807380 — CHANGES_REQUESTED** [review]
Great Work!

File: tests/test_input.py (hunk 2 above)
Comment (Bhumika-Kothwal) [reply tests/test_input.py:15]: Okay, will do the change

File: preprocessy/input/_read.py (hunk 1 above)
Comment (Bhumika-Kothwal) [reply preprocessy/input/_read.py:15]: Yes will do the changes

**Review by Saif807380 — APPROVED** [review]
LGTM!

**Pull request merged**